### PR TITLE
feat: match one dot oh release

### PR DESCRIPTION
**Pull request by jayarnielsen**
Preparation for the big 1.0.

1. Fixed various storybook errors
2. Restores commonjs 😞 
3. Reworked button and anchor to better support the new icon library (they now use an icon prop instead of nested components)

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by dprothero — APPROVED** [review]
LGTM

File: packages/match-icons/core/src/index.d.ts [hunk 1]
```diff
@@ -16,7 +16,14 @@ export interface IconProps extends React.HTMLAttributes<HTMLSpanElement> {
...
   size?: ResponsiveValue<
-    "small" | "medium" | "base" | "large" | "xLarge" | string
+    | "small"
+    | "medium"
+    | "base"
+    | "large"
+    | "xLarge"
+    | `${string}em`
+    | `${string}px`
+    | `${string}rem`
```
Comment (lacymorrow) [inline packages/match-icons/core/src/index.d.ts:26]: This is the weirdest thing I've seen in TS I think

Comment (jayarnielsen) [reply packages/match-icons/core/src/index.d.ts:26]: Pretty useful, though. Better than allowing any string at least.

**Review by lacymorrow — DISMISSED** [review]
This looks great!

I'm wondering if we should move the flag component into the "Foundation" section with the icons and logos.

**Comment by jayarnielsen** [comment]
> I'm wondering if we should move the flag component into the "Foundation" section with the icons and logos.

Yeah I was wondering about that too. Do you want to do that as part of the flag polish PR?

**Review by lacymorrow — APPROVED** [review]
LGTM++

**Pull request merged**